src/main/kotlin/org/elm/lang/core/lookup/ElmLookup.kt

```diff
@@ -52,9 +52,24 @@ object ElmLookup {
             name: String,
             module: String,
             clientLocation: ClientLocation
-    ): T? =
-            findByName<T>(name, clientLocation)
-                    .find { it.moduleName == module }
+    ): T? {
+        val results = findByName<T>(name, clientLocation).filter { it.moduleName == module }
+        return when {
+            module != "" -> results.firstOrNull()
+            else -> {
+                /*
+                The user is looking for something that belongs to an unnamed module. Generally
+                this would only happen in a toy Elm project where you haven't yet given your module
+                a name. But it is possible that a user might have an entire directory of such Elm files,
+                each without a name (see https://github.com/klazuka/intellij-elm/issues/260).
+
+                There's only one sensible way to handle it: consider it a match iff both the candidate
+                and the caller belong to the same file.
+                */
+                results.find { it.elmFile == clientLocation }
```

Comment:
This comparison against `clientLocation` is kinda dirty since there may be non-`ElmFile` implementations of the `ClientLocation` interface. But the only reason that interface exists at all is to make it easier to test: 99% of the time, it will be an `ElmFile`.